### PR TITLE
24 create trip app

### PR DESCRIPTION
**Pull request by PanLeon**
## TRELLO

* [Main Trello ticket](https://trello.com/b/BJDdC443/mytrip)

## Code reviewers

- [x] @Ashiato
- [ ] @black-art
- [x] @hrytskivr
- [x] @KarnaushenkoVO
- [ ] @PanLeon
- [ ] @spycheast
- [x] @yuriiZherebukh

### Second Level Review

- [ ] @byshovets
- [ ] @lhalam

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: myTrip/trip/models.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 from datetime import datetime
```
Comment (hrytskivr) [inline myTrip/trip/models.py:2]: unicode_literals unused import?

File: myTrip/trip/models.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
...
 from django.db import models
 from django.core.exceptions import ObjectDoesNotExist
+from registration.models import CustomUser
```
Comment (hrytskivr) [inline myTrip/trip/models.py]: use spaces to separate imports of diff type
https://www.python.org/dev/peps/pep-0008/#imports

File: myTrip/trip/models.py [hunk 3]
```diff
@@ -112,3 +118,17 @@ def delete_by_id(trip_id):
         trip = Trip.objects.get(id=trip_id)
         trip.delete()
         return None
+
+    def get_trips(user_id,page=1,step=5):
+        """
+        Returns the last 5 trips by the user
```
Comment (hrytskivr) [inline myTrip/trip/models.py:124]: if you only need fixed '5' you should delete it from arguments
and replace word 'step' below into int '5'

or you could do a function get_trips, with which user can define himself how mych trips he needs back, and just set step=5 by default.

File: myTrip/trip/views.py [hunk 4]
```diff
@@ -3,25 +3,32 @@
...
 from django.views.generic.base import View
 from .models import Trip
+from registration.models import CustomUser
 
 class TripView(View):
     """Comments view handles GET, POST, PUT, DELETE requests."""
 
-    def get(self, request, trip_id):
+    def get(self, request, trip_id=None):
         """Handles GET request"""
```
Comment (hrytskivr) [inline myTrip/trip/views.py:13]: okay, but what exactly it does?

Comment (hrytskivr) [reply myTrip/trip/views.py:13]: describe :)

File: myTrip/trip/views.py [hunk 5]
```diff
@@ -3,25 +3,32 @@
...
-        trip = Trip.get_by_id(trip_id)
-        if trip:
-            trip = trip.to_dict()
-            return JsonResponse(trip, status=200)
-        return HttpResponse(status=404)
+        if not trip_id:
+            user_id = request.user.id
+            trips = Trip.get_trips(user_id)
```
Comment (hrytskivr) [inline myTrip/trip/views.py:16]: what if there is no trips at all?

File: myTrip/tests/unittest/trip/test.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.client = Client()
+        user = CustomUser(email = 'ln@gmail.com', password = 'root')
+        user.save()
+        trip = Trip(user = user, title = 'title', description = 'description', status = 0)
+        trip.save()
+
+    def test_get(self):
```
Comment (hrytskivr) [inline myTrip/tests/unittest/trip/test.py]: organize imports
rename tests for better understanding, the test name should be self-descriptive
add docstrings
"One test should only test one thing" - L.Halamaha

check this out: https://github.com/Lv-246Python/myTrip/blob/1e009c5af925530d3fd4f4ad513f9e7af96d6760/myTrip/tests/unittests/registration/test_models.py

**Review by hrytskivr — COMMENTED** [review]
See comments, have fun :)

File: myTrip/tests/unittest/trip/test.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
+"""Testing module for checkpoint views"""
+
+import json
+from trip.models import Trip
+from datetime import datetime
+from registration.models import CustomUser
+from django.test import TestCase, Client
```
Comment (KarnaushenkoVO) [inline myTrip/tests/unittest/trip/test.py]: you should split standart packages and your own packages by empy line

File: myTrip/tests/unittest/trip/test.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+                    content_type="application/json")
+        self.assertEqual(response.status_code, 201)
+
+    def test_put(self):
+        data = {
+                    "title": "test update",
+                    "description": "some text",
+                    "status":2}
+        # trip = Trip.get_by_id(1)
+        # trip = trip.to_dict()
```
Comment (KarnaushenkoVO) [inline myTrip/tests/unittest/trip/test.py]: this is bad comment

File: myTrip/tests/unittest/trip/test.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        response = self.client.put('/api/v1/trip/1/', json.dumps(data), content_type="application/json")
+        self.assertEqual(response.status_code, 200)
+
+        response = self.client.put('/api/v1/trip/2/', json.dumps(data), content_type="application/json")
+        self.assertEqual(response.status_code, 404)
+
```
Comment (KarnaushenkoVO) [inline myTrip/tests/unittest/trip/test.py]: one test should test only one thing

File: myTrip/tests/unittest/trip/test.py [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    def test_delete(self):
+        response = self.client.delete('/api/v1/trip/1/')
+        self.assertEqual(response.status_code, 200)
+
+        response = self.client.delete('/api/v1/trip/2/')
+        self.assertEqual(response.status_code, 404)
```
Comment (KarnaushenkoVO) [inline myTrip/tests/unittest/trip/test.py]: one test should test only one thing

File: myTrip/trip/models.py [hunk 11]
```diff
@@ -36,7 +37,7 @@ def to_dict(self):
         """
         return {
             "id": self.id,
-            "user_id": self.user_id,
+            "user": self.user.id,
```
Comment (KarnaushenkoVO) [inline myTrip/trip/models.py:40]: must be "user_id"

File: myTrip/trip/models.py [hunk 12]
```diff
@@ -27,7 +28,7 @@ def to_dict(self):
             dict:
                 {
                 'id': id,
-                'user_id': user id,
+                'user_id': user,
```
Comment (KarnaushenkoVO) [inline myTrip/trip/models.py]: user_id or user, something one

File: myTrip/trip/models.py (hunk 3 above)
Comment (KarnaushenkoVO) [reply myTrip/trip/models.py:124]: Liubomyr say to use step, this is right

File: myTrip/trip/views.py (hunk 5 above)
Comment (KarnaushenkoVO) [reply myTrip/trip/views.py:16]: filter will return empty queryset, all right

**Review by oshtohryn — DISMISSED** [review]

**Review by hrytskivr — CHANGES_REQUESTED** [review]
Fix things mentioned in comments, i'll block merge for now(somehow it is available to merge whic is odd, whatever).

**Review by lhalam — COMMENTED** [review]
ToDo

**Pull request merged**